modules/hitlnext/src/backend/api.ts

```diff
@@ -180,6 +181,29 @@ export default async (bp: typeof sdk, state: StateType) => {
     })
   )
 
+  router.post(
```

Comment:
I diverge here, the `update` action is meant to update the resource properties on a 1-1 basis. The `assign` & `resolve` actions are verbs that change the state of the resource in a way that is more complex and doesn't map on a 1-1 basis with properties. Also, refactoring the `assign` and `resolve` code to be included in the `update` action would lead to a very complex/multi-scenario function that would hardly respect the principle of single responsibility.